chromium_src/third_party/blink/renderer/modules/battery/battery_manager.cc

```diff
@@ -5,27 +5,52 @@
 
 #include "third_party/blink/renderer/modules/battery/battery_manager.h"
 
+#include "third_party/blink/public/mojom/frame/lifecycle.mojom-blink.h"
 #include "third_party/blink/renderer/core/dom/dom_exception.h"
 #include "third_party/blink/renderer/core/dom/events/event.h"
 #include "third_party/blink/renderer/core/execution_context/execution_context.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
+#include "third_party/blink/renderer/core/frame/navigator.h"
 #include "third_party/blink/renderer/modules/battery/battery_dispatcher.h"
 #include "third_party/blink/renderer/platform/wtf/assertions.h"
 
 namespace blink {
 
-BatteryManager* BatteryManager::Create(ExecutionContext* context) {
-  BatteryManager* battery_manager =
-      MakeGarbageCollected<BatteryManager>(context);
-  battery_manager->UpdateStateIfNeeded();
-  return battery_manager;
+const char BatteryManager::kSupplementName[] = "BatteryManager";
+
+// static
+ScriptPromise BatteryManager::getBattery(ScriptState* script_state,
+                                         Navigator& navigator) {
+  if (!navigator.DomWindow())
+    return ScriptPromise();
+
+  // Check to see if this request would be blocked according to the Battery
```

Comment:
I'm not sure any of this stuff is actually needed since we aren't returning valid values anyway. @pes10k ?

Reply:
Agreed, we don't farble it or ever return valid values.

Reply:
Addressed in c2a768501eacacd5435b34ce6e425abf213624d2